ibis/backends/base/sql/registry/literal.py

```diff
@@ -21,7 +21,13 @@ def _boolean_literal_format(translator, op):
 
 
 def _string_literal_format(translator, op):
-    return "'{}'".format(op.value.replace("'", "\\'"))
+    return "'{}'".format(
+        op.value
+        # Escape \ first so we don't double escape other characters.
+        .replace("\\", "\\\\")
+        .replace("\n", "\\n")
+        .replace("'", "\\'")
```

Comment:
Should we use `re.escape(op.value)` instead? That'll handle other whitespace characters like `\t` as well as `\n` and `\v` and any others that are easy to miss.

Reply:
Looks like `re.escape` doesn't escape some things that we need escaped (`'`, `\n`) and does escape some things we don't want escaped (`~`, `#`, `(`, `)`, etc). I'll add a test that includes those characters.

Reply:
Updated! Turns out this fix also fixes some problems encountered in Impala regex support.